### PR TITLE
Feature #1898 tc new options

### LINKED ISSUE #1898: TCPairs and TCStat - add support for new config options and command line arguments
https://github.com/dtcenter/MET/pull/2315 added new options that will need to be supported.

## Describe the Enhancement ##
Add support in TCPairs wrapper for setting the following in the wrapped MET config file:
* diag_name - list of strings
* diag_convert_map - list of dictionaries that each contain source (string), key (list of strings), and convert (string with no quotes) - This can be handled by using the add_met_config_dict_list function found in util/met_config

Add support for setting the optional -diag {source path} command line argument in the TCPairs wrapper

Add support in TCStat wrapper for setting the following in the wrapped MET config file:
* diag_thresh_name (list of strings?)
* diag_thresh_val (list of thresholds? < or gt notation?)
* init_diag_thresh_name (list of strings?)
* init_diag_thresh_val (list of thresholds? < or gt notation?)

Add support for setting -diag_thresh and -init_diag_thresh (optional?) command line arguments in the TCStat wrapper (if needed, these may be set in the job args only)

### Time Estimate ###
~2 days

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*NOAA WPO - JTTI TCDiag Project (2770043)*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### LINKED ISSUE #1926: SeriesAnalysis - add support for setting mask dictionary
Currently the mask dictionary can only be set in a METplus config file using SERIES_ANALYSIS_MET_CONFIG_OVERRIDES. We should support setting it directly.

## Describe the Enhancement ##
Add support for using SERIES_ANALYSIS_MASK_GRID (string) and SERIES_ANALYSIS_MASK_POLY (string) to set mask.grid and mask.poly in SeriesAnalysis wrapper.

### Time Estimate ###
<1 day

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [X] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
None

### Funding Source ###
None

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required
- [X] Select **scientist(s)** or **no scientist** required

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [X] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [X] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [X] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
## Change Summary
* Added support for new tc_pairs and tc_stat arguments and config variables added in dtcenter/MET#2315
* Added support for setting mask dictionary in SeriesAnalysis to close #1926
* Made unit tests that check env vars consistent to ensure all variables are checked properly -- this uncovered some minor bugs that were fixed
* Fixed logic to set MET config list of dictionaries to include missing comma (i.e. consensus)
* Some changes from PR #1919 are also included but can be ignored

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Added new unit tests, ran TCStat and TCPairs basic use cases with added settings to ensure they are set correctly

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Ensure all automated tests pass except diffs from dtcenter/MET#2335 changing ensemble stat output

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>

- [X] Please complete this pull request review by **11/14/2022**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [X] Review the source issue metadata (required labels, projects, and milestone).
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Development** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Comment by JohnHalleyGotway** [comment]
@georgemccabe please see [this comment](https://github.com/dtcenter/MET/issues/2321#issuecomment-1311043784) about changes to TC-Pairs. These changes impact this PR by moving the `diag_name` configuration option into a larger `diag_info_map` dictionary. Recommend that we pause this PR until dtcenter/MET#2321 is completed.

**Comment by georgemccabe** [comment]
Closing until changes needed to accompany dtcenter/MET#2347 are implemented.

File: internal/tests/pytests/wrappers/tc_pairs/test_tc_pairs_wrapper.py [hunk 1]
```diff
@@ -513,23 +580,42 @@ def test_tc_pairs_loop_order_processes(metplus_config, loop_by,
...
                       f"-out {out_dir}/mlq2014121318.gfso.0104"),
                      ]
 
+    # add 2nd command for cyclone 106 unless specific cyclones are requested
+    if 'TC_PAIRS_CYCLONE' not in config_overrides:
+        expected_cmds.append(
+            (f"{app_path} {verbosity} "
+             f"-bdeck {bdeck_dir}/bmlq2014123118.gfso.0106 "
+             f"-adeck {adeck_dir}/amlq2014123118.gfso.0106{diag_arg}"
+             f" -config {config_file} "
```
Comment (JohnHalleyGotway) [inline internal/tests/pytests/wrappers/tc_pairs/test_tc_pairs_wrapper.py:600]:
```suggestion
             f"-config {config_file} "
```

**Review by JohnHalleyGotway — APPROVED** [review]
I approve of these changes. Note however that this is just based on a code review. No use case exists to demonstrate their use yet. However I believe @jvigh plans to add on in the next 6 months or so which should serve as a good test.

**Pull request merged**